src/main/java/net/mcreator/ui/modgui/ParticleGUI.java

```diff
@@ -77,6 +78,10 @@ public ParticleGUI(MCreator mcreator, ModElement modElement, boolean editingMode
 				L10N.t("elementgui.particle.visual_scale"), AbstractProcedureSelector.Side.CLIENT,
 				new JSpinner(new SpinnerNumberModel(1, 0.1, 4096, 0.1)), 0,
 				Dependency.fromString("x:number/y:number/z:number/world:world/age:number/scale:number"));
+		rotationProvider = new ProcedureSelector(this.withEntry("particle/rotation_provider"), mcreator,
+				L10N.t("elementgui.particle.rotation_provider"), ProcedureSelector.Side.CLIENT, true,
+				VariableTypeLoader.BuiltInTypes.VECTOR, Dependency.fromString("ageTicks:number")).setDefaultName(

```

Comment:
Other procedures provide much more than just age, I suggest you follow format of other procedures for particle that provide much more dependencies particle exposes